pkgs/development/python-modules/devito/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, anytree
+, nbval
+, sympy
+, scipy
+, cached-property
+, psutil
+, py-cpuinfo
+, cgen
+, click
+, multidict
+, distributed
+, pyrevolve
+, codepy
+, pytestCheckHook
+, matplotlib
+}:
+
+buildPythonPackage rec {
+  pname = "devito";
+  version = "4.6.2";
+
+  src = fetchFromGitHub {
+    owner = "devitocodes";
+    repo = "devito";
+    rev = "v${version}";
+    sha256 = "sha256-1yPKaVtxjeoANCUAqJYZndi9C/1BVVg5vuNNAb19zvs=";
+  };
+
+  postPatch = ''
+    # Small fix to avoid runtime errors (PR number 1850)
```

Comment:
Add url here: https://github.com/devitocodes/devito/pull/1850

And rather than doing this, use fetchpatch:

```
# add fetchpatch to inputs at top
...
patches = [
  # add pr url here
  (fetchpatch {
    url = "https://github.com/devitocodes/devito/commit/b46a51dfdc67e511dcd2fb3eecc4cb8f5aa9afc2.patch";
    hash = "";
  })
]